### PR TITLE
minor refactor to allow modular functions

### PR DESCRIPTION
**Pull request by Ssukriti**
### Description of the change

allows minor refactoring to allow easier integration of pretokenized input and other forms of masking
Changes added:

    1. validate_data_args function - validates the JSON formats. Can be further extended to do  any validation for pretokenized datasets can also be done in this function

    2. get_data_collator - function to return appropriate data collator as per data format .

### Related issue number

1. Related to : https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/991 - gives modules on where to add pretokenized changes
2. first step to allow masking without completion LM

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [ ] I have ensured all unit tests pass

### CODE REVIEW

File: tuning/utils/preprocessing_utils.py [hunk 1]
```diff
@@ -18,32 +18,149 @@
...
+        #     # if dataset_text_field is None and response_template is None:
+        #         # Use the seq2seq data collator; note that this automatically pads labels with -100
+        #         return DataCollatorForSeq2Seq(
+        #             tokenizer=tokenizer, padding=True, max_length=max_sequence_length
+        #         )
+        # 2. add anything needed for preprocessed input
+
+
+###################################################################################
+### The functions below are not yet used. Iterative development towards new features
```
Comment (Ssukriti) [inline tuning/utils/preprocessing_utils.py:117]: dont worry about reviewing anything below this line as it is not used yet

File: tuning/sft_trainer.py [hunk 2]
```diff
@@ -244,31 +237,14 @@ def train(
...
-            # We should do this validation up front, then do the encoding, then handle the collator
-            raise ValueError("Response template is None, needs to be set for training")
-        data_collator = DataCollatorForCompletionOnlyLM(
-            response_template_ids,
-            tokenizer=tokenizer,
-            ignore_index=configs.IGNORE_INDEX,
-        )
         packing = False
 
-    # Currently we support formatted datasets with single sequence instances.
```
Comment (Ssukriti) [inline tuning/sft_trainer.py:258]: moved to validate function

File: tuning/sft_trainer.py [hunk 3]
```diff
@@ -195,14 +196,6 @@ def train(
             }
         )
 
-    # TODO: near term - how response template ids are parsed out needs to be cleaned.
```
Comment (Ssukriti) [inline tuning/sft_trainer.py:198]: moved to get_data_collator function as we only need it for completionLM colator

**Review by kmehant — APPROVED** [review]

File: tuning/utils/preprocessing_utils.py [hunk 4]
```diff
@@ -25,25 +25,141 @@
...
 
-def validate_data_args(
+def validate_data_args(data_args: configs.DataArguments, packing: bool):
+
+    assert isinstance(
+        data_args.training_data_path, str
+    ), "Training data path has to be set and str"
+
+    # Dataset containing single sequence needs a response template for masking
+    if data_args.response_template is None and data_args.dataset_text_field is not None:
```
Comment (praveenj83) [inline tuning/utils/preprocessing_utils.py:35]: Isn't this condition fully subsumed by the next error condition? The next error condition is when packing is false and response_template is empty. This one is the same condition AND dataset_text_field is non-empty. The error will be spotted even without this condition - is the intent to provide a specific error message?

**Review by praveenj83 — COMMENTED** [review]
One minor comment on this PR

**Pull request merged**